src/common/recordbatch/src/adapter.rs

```diff
@@ -151,28 +152,29 @@ impl Stream for AsyncRecordBatchStreamAdapter {
         loop {
             match &mut self.state {
                 AsyncRecordBatchStreamAdapterState::Uninit(stream_future) => {
-                    self.state = AsyncRecordBatchStreamAdapterState::Inited(ready!(Pin::new(
-                        stream_future
-                    )
-                    .poll(cx)));
+                    self.state = match ready!(Pin::new(stream_future).poll(cx)) {
+                        Ok(stream) => AsyncRecordBatchStreamAdapterState::Ready(stream),
+                        Err(e) => AsyncRecordBatchStreamAdapterState::Failed(Some(e)),
+                    };
                     continue;
```

Comment:
Maybe we can return the error right here. And simplify the state to `AsyncRecordBatchStreamAdapterState::Failed` without an extra optional error